### ISSUE TITLE
On android 6, it is mandatory to keep saveToGallery: true, otherwise the callback won't be invoked.

### ISSUE THREAD

**Issue by briosheje** [NativeScript/nativescript-camera#88]
### If the demo apps cannot help and there is no issue for your problem, tell us about it
Please, ensure your title is less than 63 characters long and starts with a capital
letter.
On android 6, saveToGallery is mandatory (true).

### Which platform(s) does your issue occur on?
- iOS/Android/Both --- Both
- iOS/Android versions --- not relevant on ios, android 6.0
- emulator or device. What type of device? Devices: Iphone SE, Iphone X, Motorola Moto X, Nexus 5X, Samsung galaxy S7, Iphone 7.

### Please, provide the following version numbers that your issue occurs with:

- CLI: (run `tns --version` to fetch it)  3.3.0
- Cross-platform modules: (check the 'version' attribute in the
`node_modules/tns-core-modules/package.json` file in your project) ^3.4.0
- Runtime(s): (look for the `"tns-android"` and `"tns-ios"` properties in the  `package.json` file of your project) 3.3.0 (tns-android), 3.4.1 (tns-ios)
- Plugin(s): (look for the version numbers in the `package.json` file of your
project and paste your dependencies and devDependencies here)

### Please, tell us how to recreate the issue in as much detail as possible. 
Describe the steps to reproduce it.
See below

### Is there any code involved? 
 - provide a code example to recreate the problem 
 - (EVEN BETTER) provide a .zip with application or refer to a repository with application where the problem is reproducible.

I was struggling about why the callback wasn't being fired in this piece of code:

```
    console.log("taking picture!");
    camera.takePicture(<camera.CameraOptions>{
      saveToGallery: false
    })
    .then((imageAsset) => {
        console.log("Result is an image asset instance");
        this.askFilename().then((filename: string) => {
          console.log('filename chosen by the user:');
          console.log(filename);
          let imgFormat = "png";
          this.filesystem.writeImageFile(imageAsset, this.filesystem.mkdir("mydir"), filename, imgFormat).then((done: boolean) => {
            if (done) {
              this.localFilesComponent.refreshLocalData();
            }
            else {
              console.log('oh, snap.');
            }
          });
        });
    }).catch((err) => {
        console.log("Error -> " + err.message);
    });
```

Basically, the `.then(imageAsset)` was not being called **on android only**. The problem was reproduceable on Nexus 5X, Moto X but was properly working on Samsung. In iOS, it was just working.

After removing: `saveToGallery: false`, it worked perfectly on all the devices.

This shouldn't be a huge problem, the big issue, though, is that **no error was thrown and no callback was fired, the console just reported:**

```
JS: taking picture!
JS: >>> onActivityResult 1: 3453
```

**Comment by NickIliev** [comment]
Steps to reproduce - use [this demo application.](https://github.com/NickIliev/NS-Issues-2018/tree/master/nativescript-camera/issue_88) and take photos on Android API 23 and above (tested on API 27) with `saveToGallery: false`

Update: after second evaluation of this issue I am not able to reproduce the behavior with the test application (after taking a picture the promise is always successfully resolved with or without explicitly setting saveToGaleery option to `false`). The default value is `false` so you could try to simply emit the option `saveToGallery`.
Please let me know if you are still experiencing this issue with [my sample project](https://github.com/NickIliev/NS-Issues-2018/tree/master/nativescript-camera/issue_88)

**Comment by zbranzov** [comment]
I couldn't reproduce the issue using Nexus 5 Android 6.0. Since it works fine on some devices and doesn't on others I think that we should keep the current state and just emit the property as @NickIliev  said.

**Comment by briosheje** [comment]
@NickIliev I'll test the demo application on the said phone (Moto X with Android 6.0) and on a Nexus 5X again. Unluckily, in my application, I'm still experiencing the issue. I will keep you updated (with further screens eventually, if needed).

**Comment by zbranzov** [comment]
@briosheje We changed the default value to 'true' for both platforms in the latest released version 4.0.0

**Comment by briosheje** [comment]
@zbranzov Thanks. Not sure whether it's relevant, but chaning from 3.3.0 to 3.4.1 for tns-android apparently solved the issue as well, not sure whether that's intended or not but whatever, just wanted to share it.
